### PR TITLE
fix expression changed error

### LINKED ISSUE #301: [help wanted] Expression changed after it has been checked.
Hello,

I am using your component in a project and I got a recent bug report of an ExpressionChangedAfterItHasBeenCheckedError.

Normally or 99% of the time this is due to a programming bug of myself, but this time I can't find the cause of this error and I was hoping if I could get some extra information here to get this fixed.

This is a snapshot of the error thrown. I also want to point out that this happens only on a local build, not on the deployed one. To reproduce this error I need to enter some text and press spacebar sometimes.
image

### PR DESCRIPTION
**Pull request by gustavovitor**
fixes #301

### CODE REVIEW

**Comment by macjohnny** [comment]
thanks for your contribution. the visibility check of the not found message additionally requires to check whether a value was entered and whether a label is defined.
see 
https://github.com/bithost-gmbh/ngx-mat-select-search/blob/c22f84c3ffd0df08d95b998d2e34d66eb65c93d8/src/app/mat-select-search/mat-select-search.component.ts#L246-L252

**Comment by macjohnny** [comment]
a solution you could try is to modify the code to something like
```typescript
public _showNoEntriesFound$: Observable<boolean> = combineLatest([ 
   this._formControl.valueChanges, 
   this.optionsLength$ 
 ]).pipe(
   timer(), 
   map(([value, optionsLength]) => this.noEntriesFoundLabel && value 
     && optionsLength === this.getOptionsLengthOffset()) 
 ); 
```

**Comment by gustavovitor** [comment]
Thanks for explain @macjohnny
Works very well with one debounceTime(0), lying to the life cycle of Angular, haha.

File: src/app/mat-select-search/mat-select-search.component.ts [hunk 1]
```diff
@@ -247,6 +247,7 @@ export class MatSelectSearchComponent implements OnInit, OnDestroy, ControlValue
     this._formControl.valueChanges,
     this.optionsLength$
   ]).pipe(
+    debounceTime(0),
```
Comment (macjohnny) [inline src/app/mat-select-search/mat-select-search.component.ts:250]:
```suggestion
    delay(0),
```

File: src/app/mat-select-search/mat-select-search.component.ts [hunk 2]
```diff
@@ -30,8 +30,8 @@ import { MatFormField } from '@angular/material/form-field';
 import { A, DOWN_ARROW, END, ENTER, ESCAPE, HOME, NINE, SPACE, UP_ARROW, Z, ZERO, } from '@angular/cdk/keycodes';
 import { ViewportRuler } from '@angular/cdk/scrolling';
 import { LiveAnnouncer } from '@angular/cdk/a11y';
-import { BehaviorSubject, combineLatest, Observable, of, Subject } from 'rxjs';
-import { delay, filter, map, startWith, switchMap, take, takeUntil, tap } from 'rxjs/operators';
+import { BehaviorSubject, combineLatest, Observable, of, Subject, timer } from 'rxjs';
+import { debounceTime, delay, filter, map, startWith, switchMap, take, takeUntil, tap } from 'rxjs/operators';
```
Comment (macjohnny) [inline src/app/mat-select-search/mat-select-search.component.ts:34]:
```suggestion
import { delay, filter, map, startWith, switchMap, take, takeUntil, tap } from 'rxjs/operators';
```

File: src/app/mat-select-search/mat-select-search.component.ts [hunk 3]
```diff
@@ -30,8 +30,8 @@ import { MatFormField } from '@angular/material/form-field';
 import { A, DOWN_ARROW, END, ENTER, ESCAPE, HOME, NINE, SPACE, UP_ARROW, Z, ZERO, } from '@angular/cdk/keycodes';
 import { ViewportRuler } from '@angular/cdk/scrolling';
 import { LiveAnnouncer } from '@angular/cdk/a11y';
-import { BehaviorSubject, combineLatest, Observable, of, Subject } from 'rxjs';
-import { delay, filter, map, startWith, switchMap, take, takeUntil, tap } from 'rxjs/operators';
+import { BehaviorSubject, combineLatest, Observable, of, Subject, timer } from 'rxjs';
```
Comment (macjohnny) [inline src/app/mat-select-search/mat-select-search.component.ts]:
```suggestion
import { BehaviorSubject, combineLatest, Observable, of, Subject } from 'rxjs';
```

**Review by macjohnny — CHANGES_REQUESTED** [review]

**Comment by macjohnny** [comment]
@gustavovitor can you check whether this also works with the `delay(0)`? and add a comment why the operator is there.

**Comment by macjohnny** [comment]
ok so lets go with `debounceTime(0)`

File: src/app/mat-select-search/mat-select-search.component.ts (hunk 1 above)
Comment (macjohnny) [inline src/app/mat-select-search/mat-select-search.component.ts:250]:
```suggestion
    // avoid "expression has been changed" error
    debounceTime(0),
```

**Comment by gustavovitor** [comment]
sorry @macjohnny, but with debounceTime we have a new error with the clear button, I just saw it now

Just look
image

Steps:
Open the input with custom icon
Type something
Close the input clicking outside

**Comment by macjohnny** [comment]
@gustavovitor thanks for the analysis. do you have time to look into this as well?
reopened #301

**Comment by gustavovitor** [comment]
Yes, tonight I can see that. @macjohnny

**Pull request merged**